src/platforms/Token/WithdrawPoolModal.tsx

```diff
@@ -0,0 +1,146 @@
+import { WalletTag } from "components/[guild]/crm/Identities"
+import CopyableAddress from "components/common/CopyableAddress"
+import useToast from "hooks/useToast"
+import { useTokenRewardContext } from "platforms/Token/TokenRewardContext"
+import { formatUnits } from "viem"
+import { useAccount } from "wagmi"
+import { Chains } from "wagmiConfig/chains"
+import usePool from "./hooks/usePool"
+import useWithdrawPool from "./hooks/useWithdrawPool"
+
+const WithdrawPoolModal = ({
+  isOpen,
+  onClose,
+  onSuccess,
+}: {
+  isOpen: boolean
+  onClose: () => void
+  onSuccess: () => void
+}) => {
+  const {
+    token: {
+      data: { decimals, symbol },
+    },
+    guildPlatform: {
+      platformGuildData: { chain, poolId },
+    },
+  } = useTokenRewardContext()
+
+  const { data: poolData, refetch } = usePool(chain, BigInt(poolId))
+
+  const [owner, , , poolBalance] = poolData || []
+  const balance = poolBalance ? Number(formatUnits(poolBalance, decimals)) : 0
+
+  const { chainId } = useAccount()
+  const isOnCorrectChain = Chains[chain] === chainId
+
+  const toast = useToast()
+
+  const { onSubmitTransaction: onSubmitWithdraw, isLoading: withdrawIsLoading } =
+    useWithdrawPool(chain, BigInt(poolId), () => {
+      toast({
+        status: "success",
+        title: "Success",
+        description: "Successfully withdrawed all funds from the pool!",
+      })
+      onClose()
+      refetch()
+      onSuccess()
+    })
+
+  const handleClose = () => {
+    onClose()
+  }
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={handleClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalCloseButton />
+          <ModalHeader>
+            <Text>Withdraw pool</Text>
+          </ModalHeader>
+
+          <ModalBody>
+            <Stack gap={5}>
+              <Stack gap={1}>
+                <HStack>
+                  <Text fontWeight={"semibold"} fontSize="sm">
+                    Balance
+                  </Text>
+                  <Text ml={"auto"} fontSize="sm">
+                    {balance} {symbol}
+                  </Text>
+                </HStack>
+                <HStack>
+                  <Text fontWeight={"semibold"} fontSize={"sm"}>
+                    Owner
+                  </Text>{" "}
+                  <WalletTag ml={"auto"}>
+                    <CopyableAddress address={owner} fontSize="sm" />
+                  </WalletTag>
+                </HStack>
+              </Stack>
+              <Divider />
+
+              {balance === 0 ? (
+                <>
+                  <Alert status="info">
+                    <AlertIcon mt={0} /> There are currently no funds available to
+                    withdraw from the reward pool.
+                  </Alert>
+                </>
+              ) : (
+                <>
+                  <Alert status="warning">
+                    <AlertIcon mt={0} /> Are you sure you want to withdraw all funds
```

Comment:
I think this shouldn't be a question here, we should just inform the user that "If you withdraw all funds, no further rewards can be claimed until the pool is funded again." (or something similar, I'm not good in copywriting). What do you think?